packages/checkout-storefront/src/components/SelectBox/SelectBox.tsx

```diff
@@ -1,31 +1,42 @@
 import clsx from "clsx";
 import { Children, Classes } from "@/checkout-storefront/lib/globalTypes";
-import { FC, HTMLAttributes } from "react";
+import { HTMLAttributes } from "react";
+import { useFormContext } from "@/checkout-storefront/hooks/useForm";
+import { useField } from "formik";
 
-export interface SelectBoxProps
+export interface SelectBoxProps<TFieldName extends string>
   extends Classes,
     Children,
     Omit<HTMLAttributes<HTMLInputElement>, "children"> {
-  value: string;
-  selectedValue: string | undefined;
   disabled?: boolean;
+  name: TFieldName;
+  value: string;
 }
 
-export const SelectBox: FC<SelectBoxProps> = ({
+export const SelectBox = <TFieldName extends string>({
   children,
-  value,
-  selectedValue,
   className,
-  id,
   disabled = false,
-  ...rest
-}) => {
-  const selected = selectedValue === value;
+  name,
+  value,
+  id,
+}: SelectBoxProps<TFieldName>) => {
+  const identifier = id || value;
```

Comment:
`Currently, when you pick an option in SelectBox, its id changes.` - how did you come to this conclusion? 
As to the latter, we use id as prio here specifically to avoid conflict, so when we have two lists with the same elements (like addresses) we give them custom ids (like e.g `shippingAddress-1234`). In case we have singlar lists e.g delivery methods we'd use values as id's anyway, so I don't think making it mandatory to put in jsx instead of here makes any real difference

Reply:
Picking an option changes the value. Value is used as id. Thus, id changes when an option is picked.

Reply:
Have you checked the code for such a change? We're passing some sort of static `value` to these select boxes, that being either id (that comes from the api) or prefix + id (as above).

Reply:
Why is `value` static? `value` is the currently selected option, it shouldn't be static

Reply:
Because a) it's input type radio underneath so we're following the convention of html radios and b) Formik also passes values to fields as a static string so why do you think it shouldn't be so?

Reply:
Why SelectBox is a single radio input? 🤔  I'd guess it suggests it's a… select.